src/App.js

```diff
@@ -1,16 +1,31 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [likedMessages, setLikedMessages] = useState(0);
+
+  const likeMessage = () => {
+    setLikedMessages(likedMessages => likedMessages + 1);
+  };
+
+  const unlikeMessage = () => {
+    setLikedMessages(likedMessages => likedMessages - 1);
+  };
```

Comment:
You can actually keep track of the number of likes across all the messages without using state, thus we should prefer to not use state to reduce the complexity of the project. 

Instead, you can create a function that iterates over all the chats and adds up the number of likes, something like: 

```javascript
const getHeartCount = () => {
    return chatMessages.reduce((total, chat) => {
      return chat.liked ? total + 1 : total;
    }, 0)
};

// then on line 21, you can write this:
<h2>{getHeartCount()} ❤️s</h2>
```

This approach for calculating the total likes across all messages, would require you to keep track of the messages as state though and you'd need an event handler for updating when a message is liked. I'll leave more details about that in another comment.